Draft Octane plan blog post

jenweber opened the pull request:
This article summarizes the plan for releasing Octane, and provides some guidance for early adopters.

jenweber committed:
Octane plan blog post

MelSumner reviewed (approved):
Thank you for working on this! 👍

tomdale commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,249 @@
+
+The Ember community is wrapping up the [Octane](https://emberjs.com/editions/) edition, and we expect to get it over the finish line over the next three releases!
+
+- The latest release of Ember is 3.12, which is a an LTS (long term support) candidate. As an LTS, it does not have any significant new features, but it does have many of Octane's features! Check out the [release blog posts](https://blog.emberjs.com) to learn about them.
```
Is it important to state that the LTS has many Octane features? Saying it "does not have any significant new features" but also "it does have many of Octane's features" was a little confusing for me and seemed to imply that Octane itself doesn't have any significant new features.

jenweber replied:
Simplified to:


- The next release of Ember is 3.12, which is a an LTS (long term support) candidate. Check out the [release blog posts](https://blog.emberjs.com) to learn which of Octane's features are already available in stable releases.

tomdale commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,249 @@
+
+- The latest release of Ember is 3.12, which is a an LTS (long term support) candidate. As an LTS, it does not have any significant new features, but it does have many of Octane's features! Check out the [release blog posts](https://blog.emberjs.com) to learn about them.
+- Ember 3.13 will be feature-complete for Octane, and apps using stable Ember can opt in to Octane. At this point, we're still wrapping up polish, especially in the codemods, inspector and guides.
+- In Ember 3.14, Octane mode will be the primary, recommended way to use Ember. Ember 3.14 will default new apps to Octane mode, and it will be the recommended mode. The guides and tutorials will document Octane mode and Octane idioms, and codemods to help users migrate to Octane idioms will be available.
```
"Octane mode" means specifically enabling the right optional features?

tomdale replied:
This part was a little confusing for me because so far in the post, "Octane mode" has not been defined (as a separate concept from just "Octane"). Will the concept of a "mode" be meaningful to an Ember developer reading this post? Where would they have heard this term before?

jenweber replied:
Edited to remove "mode"

- The next release of Ember is 3.12, which is a an LTS (long term support) candidate. Check out the [release blog posts](https://blog.emberjs.com) to learn which of Octane's features are already in stable releases.
- Ember 3.13 will be feature-complete for Octane, and apps using stable Ember can opt in. At this point, we're still wrapping up polish, especially in the codemods, inspector and guides.
- In Ember 3.14, Octane will be the primary, recommended way to use Ember. In Ember 3.14, new apps will have Octane's optional features enabled by default. The guides and tutorials will show Octane examples, and codemods will be available to help users migrate to Octane idioms.

tomdale commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,249 @@
+- Ember 3.13 will be feature-complete for Octane, and apps using stable Ember can opt in to Octane. At this point, we're still wrapping up polish, especially in the codemods, inspector and guides.
+- In Ember 3.14, Octane mode will be the primary, recommended way to use Ember. Ember 3.14 will default new apps to Octane mode, and it will be the recommended mode. The guides and tutorials will document Octane mode and Octane idioms, and codemods to help users migrate to Octane idioms will be available.
+
+**Octane features are opt-in**, and this will continue to be true for the rest of the 3.x series. As semver would suggest, all releases until 4.0 are backwards-compatible with 3.x.
```
s/semver/SemVer/

tomdale commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,249 @@
+---
+
+The Ember community is wrapping up the [Octane](https://emberjs.com/editions/) edition, and we expect to get it over the finish line over the next three releases!
+
```
I think it would be helpful to have a one- or two-sentence description of Octane near the introduction. The post currently dives in assuming people already know what Octane is.

jenweber replied:
>The Ember community is wrapping up the work for [Octane](https://emberjs.com/editions/), and we expect to get it over the finish line over the next three releases!
Ember Octane updates Ember with modern, streamlined components and stage management that makes it fun to build components. It has seamless interoperability with existing apps, letting teams migrate at their own pace.

jenweber replied:
This is very good feedback, thank you!

TotesOates commented on source/2019-08-15-octane-release-plan.md:
````diff
@@ -0,0 +1,249 @@
+
+```handlebars
+<Input @value={{this.name}} />
+<LinkTo @route={{this.routeName}}>Some content<LinkTo>
````
Missing slash on the closing `</LinkTo>` tag

jenweber replied:
good catch, thanks!

ijlee2 commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,249 @@
+
+The Ember community is wrapping up the [Octane](https://emberjs.com/editions/) edition, and we expect to get it over the finish line over the next three releases!
+
+- The latest release of Ember is 3.12, which is a an LTS (long term support) candidate. As an LTS, it does not have any significant new features, but it does have many of Octane's features! Check out the [release blog posts](https://blog.emberjs.com) to learn about them.
```
```suggestion
- The latest release of Ember is 3.12, which is an LTS (long term support) candidate. As an LTS, it does not have any significant new features, but it does have many of Octane's features! Check out the [release blog posts](https://blog.emberjs.com) to learn about them.
```

tomdale commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,249 @@
+
+### application-template-wrapper
+
+In classic Ember, your entire application is automatically wrapped in a `<div>` with the class `ember-application`. Octane applications do not automatically insert this uneccessary `<div>`.
```
s/uneccessary/unnecessary/ (missing 'n')

ijlee2 commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,249 @@
+
+### Consider trying Octane features that have already landed
+
+The rest of the sections below cover optional features in Octane that you can consider using now. These features are not interdependent. You can opt into them one at a time, in any order.
```
I'd avoid using a double negative.
```suggestion
The rest of the sections below cover optional features in Octane that you can consider using now. These features are independent of each other. You can opt into them one at a time, in any order.
```

tomdale commented on source/2019-08-15-octane-release-plan.md:
````diff
@@ -0,0 +1,249 @@
+
+```sh
+$ ember feature:enable template-only-glimmer-component
+```
````
Is there a migration guide or blog post for this optional feature we can link people to?

jenweber replied:
Not that I know of.

jenweber committed:
make the linter happy, maybe

jenweber committed:
Update source/2019-08-15-octane-release-plan.md

Co-Authored-By: Isaac Lee <16869656+ijlee2@users.noreply.github.com>

jenweber committed:
Update source/2019-08-15-octane-release-plan.md

Co-Authored-By: Isaac Lee <16869656+ijlee2@users.noreply.github.com>

jenweber committed:
apply feedback

jenweber committed:
Merge branch 'octane-timeline' of github.com:ember-learn/ember-blog into octane-timeline

acorncom commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,252 @@
+
+In classic Ember, jQuery is automatically included, and `this.$` inside of component classes uses jQuery to select from the DOM. Ember Octane does not include jQuery automatically, nor does it include any direct jQuery integration in the component API.
+
+Ember's internals no longer depend on jQuery. If your app still uses jQuery directly and you want to continue using it, you can add `@ember/jquery` to your `package.json` and disable this classic feature. You will still be able to use all the other features of Octane.
```
Disable or enable this classic feature?

Seems like what we're opting for here is "and keep using this classic feature"

jenweber replied:
This feature has reverse boolean logic. It is unfortunate. Enabling/disabling does the opposite of what you would think.

acorncom commented on source/2019-08-15-octane-release-plan.md:
```diff
@@ -0,0 +1,252 @@
+
+## Using Octane idioms today
+
+Because many of Octane's features have a already landed, you can start trying out these new features today.
```
```suggestion
Because many of Octane's features have already landed, you can start trying out these new features today.
```

jenweber committed:
update intro

jenweber committed:
linting, again

jenweber committed:
Update source/2019-08-15-octane-release-plan.md

Co-Authored-By: David Baker <acorncom@users.noreply.github.com>

acorncom reviewed (approved):
Looks great. One not-sure-it-needs-to-be-handled thought. Idiom and idiomatic are used heavily through the blog post, but I'm not sure we use that much in the guides. Are we at all concerned about that tripping up non-native English speakers here?

jenweber commented:
@acorncom that is very good feedback and yes, we should define idiom/idiomatic. I will put something small.

jenweber commented:
> When we say "Octane idioms," we mean the syntax and patterns that Octane introduces.

I removed unnecessary usages of the word idiom and added this definition the first time we use it.

jenweber committed:
define idioms and reduce usage

jenweber committed:
Merge branch 'octane-timeline' of github.com:ember-learn/ember-blog into octane-timeline

jenweber committed:
fix headers

jenweber commented:
I also edited to remove a redundant section header, and replaced it with a more general description.

jenweber committed:
once more, with feeling

jenweber committed:
fix channel name

jenweber committed:
clarify opening statement

jenweber committed:
existing -> exciting

jenweber commented:
Since this blog post was significantly reviewed and edited by multiple people before it was a PR, and now that I have incorporated the feedback, I'm going to merge! I'm excited to get this info available!
